index.js

```diff
@@ -8,30 +9,47 @@ function emptyDir(dir, filter, cb) {
     filter = null;
   }
 
-  if (typeof cb !== 'function') {
+  if (cb && typeof cb !== 'function') {
     throw new TypeError('expected callback to be a function');
   }
 
-  if (Array.isArray(dir)) {
-    cb(null, isEmpty(dir, filter));
-    return;
-  }
-
-  if (typeof dir !== 'string') {
-    cb(new TypeError('expected a directory or array of files'));
-    return;
+  if (!Array.isArray(dir) && typeof dir !== 'string') {
+    throw new TypeError('expected a directory or array of files')
```

Comment:
please add semicolons, following the previous conventions. thx

Reply:
Sorry about that. Will do.